From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Charles Kerr <charles@charleskerr.com>
Date: Mon, 29 Jul 2024 18:12:40 -0500
Subject: refactor: use non-deprecated openAppliationAtURL

Use openAppliationAtURL instead of launchApplicationAtURL,
which was deprecated in macOS 11.

This patch can be removed if/when upstream removes its use of
deprecated macOS API.

diff --git a/Squirrel/ShipIt-main.m b/Squirrel/ShipIt-main.m
index 671f8fa2104df85046ff813d4b824a65caae502f..d9efbe5f8337921901509d00fa022b3c41489bf6 100644
--- a/Squirrel/ShipIt-main.m
+++ b/Squirrel/ShipIt-main.m
@@ -186,12 +186,8 @@ static void installRequest(RACSignal *readRequestSignal, NSString *applicationId
 							NSLog(@"New ShipIt exited");
 						} else {
 							NSLog(@"Attempting to launch app on lower than 11.0");
-							if (![NSWorkspace.sharedWorkspace launchApplicationAtURL:bundleURL options:NSWorkspaceLaunchDefault configuration:@{} error:&error]) {
-								NSLog(@"Could not launch application at %@: %@", bundleURL, error);
-								return;
-							}
-
-							NSLog(@"Application launched at %@", bundleURL);
+							NSLog(@"Could not launch application at %@", bundleURL);
+							return;
 						}
 					}];
 			}
@@ -235,13 +231,16 @@ int main(int argc, const char * argv[]) {
 
 		if (strcmp(jobLabel, [launchSignal UTF8String]) == 0) {
 			NSLog(@"Detected this as a launch request");
-			NSError *error;
-			if (![NSWorkspace.sharedWorkspace launchApplicationAtURL:shipItStateURL options:NSWorkspaceLaunchDefault configuration:@{} error:&error]) {
-				NSLog(@"Could not launch application at %@: %@", shipItStateURL, error);
-			} else {
-				NSLog(@"Successfully launched application at %@", shipItStateURL);
-			}
-			exit(EXIT_SUCCESS);
+			[NSWorkspace.sharedWorkspace openApplicationAtURL:shipItStateURL
+					configuration:NSWorkspaceOpenConfiguration.configuration
+					completionHandler:^(NSRunningApplication* app, NSError* error) {
+				if (error) {
+					NSLog(@"Could not launch application at %@: %@", shipItStateURL, error);
+				} else {
+					NSLog(@"Successfully launched application at %@", shipItStateURL);
+				}
+				exit(EXIT_SUCCESS);
+			}];
 		} else {
 			NSLog(@"Detected this as an install request");
 			installRequest([SQRLShipItRequest readUsingURL:[RACSignal return:shipItStateURL]], @(jobLabel));
